### PR TITLE
SingularKey Error Handling

### PR DESCRIPTION
**Pull request by kristinaasalgado**
Ended up making a new file to handle parsing SingularKey errors which we can maybe also use in password and profile updates since those will also be utilizing SingularKey.

### CODE REVIEW

File: bxhealth/src/components/ModalRegisterCaregiver/ModalRegisterCaregiver.js [hunk 1]
```diff
@@ -66,9 +68,17 @@ class ModalRegisterCaregiver extends React.Component {
...
+            console.log("Error", error.message);
+            this.setState({
+              haveError: true,
+              errorTitle: error.title,
+              errorMsg: error.message
+            })
+          };
         });
-    } else console.log("Registration error 2.")
+    } else console.log("Registration form not valid.")
```
Comment (jon-oblander) [inline bxhealth/src/components/ModalRegisterCaregiver/ModalRegisterCaregiver.js:81]: test

File: bxhealth/src/components/Utils/SingularKeyErrors.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+class SingularKeyErrors {    
+    constructor() {
+        this.envVars = window._env_;
```
[the remark this answers is not in the document]
Comment (kristinaasalgado) [reply bxhealth/src/components/Utils/SingularKeyErrors.js:3]: Removed!

File: bxhealth/src/components/Utils/SingularKeyErrors.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    @param {object} response Response returned from SingularKey to be compared and/or parsed.
+    @return {object} User-friendly error title and message.
+
+     */
+    errorMessage(response) {
+        const body = response.additionalProperties.body;
+        const error = JSON.parse(body);
+        const errorString = error.details[0]?.message ?? "";
+
+        var uid = "";
```
[the remark this answers is not in the document]
Comment (kristinaasalgado) [reply bxhealth/src/components/Utils/SingularKeyErrors.js:19]: Oops, yes. Updated! Thank you.

File: bxhealth/src/components/Utils/SingularKeyErrors.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    /**
+    Generate Error Message:
+    Parse errors from SingularKey so they are better formatted for user displays.
+
+    @param {object} response Response returned from SingularKey to be compared and/or parsed.
+    @return {object} User-friendly error title and message.
+
+     */
+    errorMessage(response) {
```
[the remark this answers is not in the document]
Comment (kristinaasalgado) [reply bxhealth/src/components/Utils/SingularKeyErrors.js:14]: Chatted with Jon about this one - leaving as is for now but will continue to assess as this file gets used.

**Review by jon-oblander — APPROVED** [review]

**Pull request merged**